Drake Version Update to Enable Building of Branch

sharanyashastry opened the pull request:
Upgraded Drake from version 1.22.0 to 1.33.0. This is the latest version that built without needing any additional changes.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/DAIRLab/dairlib/369)
<!-- Reviewable:end -->

sharanyashastry committed:
Upgraded drake to newest version that required no other changes to build the branch.

ebianchi commented:
Testing a fresh install, I get the following "issues," though everything still builds properly:

- Running the installation script `sudo ./install/install_prereqs_jammy.sh` gets the error (related to #367):
     - FIX:  This will be fixed if the changes to `install_prereqs_jammy.sh` are brought into this PR (as they should).
```
./source_distribution/install_prereqs.sh: line 116: ./source_distribution/install_bazelisk.sh: No such file or directory
```

- Bazel building with `bazel build ...` gives the warning that it's creating an empty file (that git then marks as an untracked file):
     - FIX:  Two options here:  migrate from `WORKSPACE.bazel` to `MODULE.bazel`, or add the line `common --noenable_bzlmod` to `.bazelrc` to disable Bzlmod.
```
WARNING: --enable_bzlmod is set, but no MODULE.bazel file was found at the workspace root. Bazel will create an empty MODULE.bazel file. Please consider migrating your external dependencies from WORKSPACE to MODULE.bazel. For more details, please refer to https://github.com/bazelbuild/bazel/issues/18958.
```

ebianchi commented on WORKSPACE:
```diff
@@ -11,12 +11,12 @@ workspace(name = "dairlib")
 # Before changing the COMMIT, temporarily uncomment the next line so that Bazel
 # displays the suggested new value for the CHECKSUM.
-#DRAKE_CHECKSUM = "0" * 64
+# DRAKE_CHECKSUM = "0" * 64
```
Not a big deal, but there's no point in changing this line.

sharanyashastry replied:
Fixed this

sharanyashastry committed:
Removing white space change.

sharanyashastry committed:
Adding install prereqs changes to this PR.

sharanyashastry committed:
Added noenable_bzlmod flag in bazelrc.

sharanyashastry commented:
Updated:

1. Brought the changes to `install_prereqs_jammy.sh` into this PR.
2. Added the line `common --noenable_bzlmod` to the `.bazelrc` to disable Bzlmod.

Tested a clean build that built with neither of the above warnings.

ebianchi commented on install/install_prereqs_jammy.sh:
```diff
@@ -23,14 +23,16 @@ touch install_prereqs_user_environment.sh
 wget "https://raw.githubusercontent.com/RobotLocomotion/drake/${DRAKE_COMMIT}/setup/ubuntu/source_distribution/packages-${ubuntu_codename}-clang.txt"
 wget "https://raw.githubusercontent.com/RobotLocomotion/drake/${DRAKE_COMMIT}/setup/ubuntu/source_distribution/packages-${ubuntu_codename}-test-only.txt"
+wget "https://raw.githubusercontent.com/RobotLocomotion/drake/${DRAKE_COMMIT}/setup/ubuntu/source_distribution/install_bazelisk.sh"
+wget "https://raw.githubusercontent.com/RobotLocomotion/drake/${DRAKE_COMMIT}/setup/ubuntu/source_distribution/install_bazel.sh"
```
This line gives an error:
```
--2025-05-19 16:11:57--  https://raw.githubusercontent.com/RobotLocomotion/drake/v1.33.0/setup/ubuntu/source_distribution/install_bazel.sh
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 2606:50c0:8000::154, 2606:50c0:8003::154, 2606:50c0:8001::154, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|2606:50c0:8000::154|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2025-05-19 16:11:58 ERROR 404: Not Found.
```

However this isn't consequential given `install_bazelisk.sh` does successfully install.  This might've been written this way to accommodate many versions of Drake that have only one or the other.  Seems fine to leave to me.

ebianchi reviewed (approved):
Confirmed the build worked without issues, and no files were inadvertently added.
